backend/src/main/java/fk/prof/backend/model/aggregation/impl/AggregationWindowLookupStoreImpl.java

```diff
@@ -0,0 +1,42 @@
+package fk.prof.backend.model.aggregation.impl;
+
+import com.google.common.base.Preconditions;
+import fk.prof.backend.aggregator.AggregationWindow;
+import fk.prof.backend.model.aggregation.AggregationWindowLookupStore;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AggregationWindowLookupStoreImpl implements AggregationWindowLookupStore {
+  private Map<Long, AggregationWindow> windowLookup = new ConcurrentHashMap<>();
+
+  //NOTE: Called on http event loop
+  @Override
+  public AggregationWindow getAssociatedAggregationWindow(long workId) {
+    return this.windowLookup.get(workId);
+  }
+
+  //NOTE: Called always by backend daemon thread and not on http event loop
+  @Override
+  public void associateAggregationWindow(long[] workIds, AggregationWindow aggregationWindow)
+      throws IllegalStateException {
+    Preconditions.checkNotNull(aggregationWindow);
```

Comment:
Aggregation window is being inserted in a map. Depending on map's behavior, null may or may not be allowed (interface does not prohibit, hashmap allows). Presently concurrenthashmap is used which will throw npe but we do not want the code to work only with a specifc impl. We definitely do not want nulls to be inserted in the map, so a null check beforehand makes sense here. Making the other param final